router/src/main/java/io/micronaut/web/router/CustomMethodRoute.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.web.router;
+
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import io.micronaut.core.type.Argument;
+import io.micronaut.core.type.ReturnType;
+import io.micronaut.http.HttpMethod;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.uri.UriMatchTemplate;
+import io.micronaut.http.uri.UriMatchVariable;
+import io.micronaut.inject.ExecutableMethod;
+
+/**
+ * A special class that routes everything non-standard for http.
+ * @author spirit-1984
+ * @since 1.2.1
+ */
+class CustomMethodRoute implements UriRoute {
```

Comment:
Why is this class necessary?

Reply:
Ok, so I did this intentionally, cause I did not want to mess around with internals of DefaultUriRoute. The logic is that this class is used only for non-standard http methods, and the standard are working with the same code of DefaultUriRoute. I agree, this is overengineering, and too much worry. I'll fix that and modify DefaultUriRoute